### PR TITLE
feat: update Lua tree generation script & patch file for Kong 3.1.0-0 Luarock

### PR DESCRIPTION
**Pull request by omegabytes**

### CODE REVIEW

**Comment by omegabytes** [comment]
Need to add support for/replace ngx.base64 (probably alias it [like we did with ngx.re.match](https://github.com/Kong/goks/pull/26))

File: lua-tree/share/lua/5.1/kong/tools/utils.lua [hunk 1]
```diff
@@ -1263,4 +1264,101 @@ end
...
+
+    return value
+  end
+end
+_M.try_decode_base64 = try_decode_base64
+
+
+local sha256_bin
+do
+  local digest = require "resty.openssl.digest"
```
Comment (omegabytes) [inline lua-tree/share/lua/5.1/kong/tools/utils.lua:1304]: This relies on `ffi` and will need to be removed/replaced (see https://github.com/Kong/goks/blob/main/patches/lua-tree.patch#L68)

**Comment by omegabytes** [comment]
> Need to add support for/replace ngx.base64 (probably alias it [like we did with ngx.re.match](https://github.com/Kong/goks/pull/26))

I'm not so sure about this after looking at how base64 us being used across the project. Why is decode_base64 imported directly but decode_base64url imports ngx.base64 first? If I update the imports to match then things work correctly. [One example here](https://github.com/Kong/kong/blob/master/kong/tools/utils.lua#L1474-L1475)
```
  local decode_base64    = ngx.decode_base64
  local decode_base64url = require "ngx.base64".decode_base64url
```
To resolve, we could fix the import in kong, or I can patch it in goks. I wrote some throwaway logic that aliases `decode_base64` and `decode_base64url` methods using the go encoding lib [here](https://github.com/Kong/goks/blob/a7090dc97d59cdd25e3ba74328839d03d0bb8256/lualibs/go/ngx/base64.go).

**Comment by mikefero** [comment]
@omegabytes With 3.2.0 on the horizon I am going to close this PR in favor of grabbing the newest updates after that release occurs.

**Pull request closed without merging**